client/asset/eth/eth.go

```diff
@@ -579,6 +580,29 @@ func genWalletSeed(entropy []byte) ([]byte, error) {
 	return bip39.NewSeed(mnemonic, ""), nil
 }
 
+func privKeyFromSeed(seed []byte) (pk []byte, zero func(), err error) {
+	walletSeed, err := genWalletSeed(seed)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer encode.ClearBytes(walletSeed)
+
+	extKey, err := keygen.GenDeepChild(walletSeed, seedDerivationPath)
+	if err != nil {
+		return nil, nil, err
+	}
+	// defer extKey.Zero()
+	pk, err = extKey.SerializedPrivKey()
+	if err != nil {
+		extKey.Zero()
+		return nil, nil, err
+	}
+	return pk, func() {
+		extKey.Zero()
+		defer encode.ClearBytes(pk)
```

Comment:
Not new in this commit, but turns out we shouldn't be doing `encode.ClearBytes(pk)`.  The docs for `SerializedPrivKey` say `The returned bytes must not be modified.` and it is because it is a shallow copy of the byte slice (`extKey.Zero()` handles it).

Reply:
I noticed it was a shallow copy, but I didn't want to assume anything. Will just use `Zero`.

Reply:
Understandable, but since `SerializedPrivKey` docs are explicit about it, I think we should not zero the slice.